File: PointOfSaleTerminal/Java/src/com/sandbox/Item.java
package com.sandbox;

public class Item
{
    private String productCode;
    private Double singlePrice;
    private Double volumePrice;
    private Integer volumeQuantity;
    
    public Item( String productCodeIn, Double singlePriceIn, Double volumePriceIn, Integer volumeQuanityIn )
    {
        productCode = productCodeIn;
        singlePrice = singlePriceIn;
        volumePrice = volumePriceIn;
        volumeQuantity = volumeQuanityIn;
    }
    
    public String getProductCode()
    {
        return productCode;
    }
    
    public Double getSingleUnitPrice()
    {
        return singlePrice;
    }
    
    public Double getVolumeUnitPrice()
    {
        return volumePrice;
    }
    
    public Integer getVolumeQuantity()
    {
        return volumeQuantity;
    }
}

File: PointOfSaleTerminal/C++/Item.cpp
#include "Item.h"

Item::Item( std::string productCodeIn, double singlePriceIn, double volumePriceIn, uint32_t volumeQuantityIn ) :
    _productCode( productCodeIn ), _singlePrice( singlePriceIn ), _volumePrice( volumePriceIn ), _volumeQuantity( volumeQuantityIn )
{

}

double Item::getSingleUnitPrice() const
{
    return _singlePrice;
}

double Item::getVolumeUnitPrice() const
{
    return _volumePrice;
}

uint32_t Item::getVolumeQuantity() const
{
    return _volumeQuantity;
}
